test/org/javarosa/xpath/expr/XPathFuncExprRandomizeTest.java

```diff
@@ -0,0 +1,179 @@
+
+        assertTrue(nodesEqualInOrder(choices1, choices2));
+    }
+
+    @Test
+    public void the_same_seeded_field_in_different_instances_from_deserialized_forms_gets_the_same_order_of_choices() throws IOException, DeserializationException {
+        initializeNewInstance(formDef);
+        List<SelectChoice> choices1 = getSelectChoices(formDef, "/randomize/seededFruit2");
+
+        FormDef formDefAfterSerialization = serializeAndDeserializeForm(formDef);
+
+        initializeNewInstance(formDefAfterSerialization);
+        List<SelectChoice> choices2 = getSelectChoices(formDefAfterSerialization, "/randomize/seededFruit2");
+
+        assertTrue(nodesEqualInOrder(choices1, choices2));
+    }
+
+    private FormDef serializeAndDeserializeForm(FormDef formDef) throws IOException, DeserializationException {
+        // Initialize serialization
+        PrototypeManager.registerPrototypes(JavaRosaCoreModule.classNames);
+        PrototypeManager.registerPrototypes(CoreModelModule.classNames);
+        new XFormsModule().registerModule();
+
+        // Serialize form in a temp file
+        Path tempFile = createTempFile("javarosa", "test");
+        formDef.writeExternal(new DataOutputStream(newOutputStream(tempFile)));
+
+        // Create an empty FormDef and deserialize the form into it
+        FormDef deserializedFormDef = new FormDef();
+        logger.debug("Calling readExternal on new FormDef");
+        deserializedFormDef.readExternal(
+            new DataInputStream(newInputStream(tempFile)),
+            PrototypeManager.getDefault()
+        );
+
+        delete(tempFile);
+        return deserializedFormDef;
+    }
+
+    private static void initializeNewInstance(FormDef formDef) {
+        formDef.initialize(true, new InstanceInitializationFactory());
+    }
+
+    private List<SelectChoice> getSelectChoices(FormDef formDef, String ref) {
+        FormIndex formIndex = getFormIndex(formDef, absoluteRef(ref));
+        FormEntryPrompt formEntryPrompt = new FormEntryPrompt(formDef, formIndex);
+        return formEntryPrompt.getSelectChoices();
+    }
+
+    private FormIndex getFormIndex(FormDef formDef, TreeReference ref) {
+        for (int localIndex = 0, lastIndex = formDef.getChildren().size(); localIndex < lastIndex; localIndex++)
+            if (formDef.getChild(localIndex).getBind().getReference().equals(ref))
+                return new FormIndex(localIndex, 0, ref);
+        throw new IllegalArgumentException("Reference " + ref + " not found");
+    }
+
+    private static boolean nodesEqualInOrder(List<SelectChoice> left, List<SelectChoice> right) {
+        if (left.size() != right.size())
+            return false;
+
+        for (int i = 0; i < left.size(); i++)
+            if (!left.get(i).getValue().equals(right.get(i).getValue()))
+                return false;
+
+        return true;
+    }
+
+    private static TreeReference absoluteRef(String path) {
```

Comment:
I tried to solve the same problem at https://github.com/opendatakit/javarosa/commit/ee5527fd3c2d49549e288de6beaab5df48068be8. I think your approach always works, right? If so, it's simpler than mine and is a good example of something we can set up in some kind of test helper.

Reply:
Yes, I'm sure it works at least in the context of this test. If you think this is useful, I can extract that into a helper class.